portfolio/src/main/java/com/google/sps/servlets/TitanicDataServlet.java

```diff
@@ -0,0 +1,72 @@
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Returns titanic survivor data by class in a hash map*/
+@WebServlet("/titanic-data")
+public class TitanicDataServlet extends HttpServlet {
+
+  private LinkedHashMap<String, Double> titanicSurvivors = new LinkedHashMap<>();
+
+  @Override
+  public void init() {
+    Double firstPercent = 0.0;
+    Double secondPercent = 0.0;
+    Double thirdPercent = 0.0;
+
+    Double firstTotal = 0.0;
+    Double secondTotal = 0.0;
+    Double thirdTotal = 0.0;
+
+    Double firstSurvived = 0.0;
+    Double secondSurvived = 0.0;
+    Double thirdSurvived = 0.0;
+
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream(
+        "/WEB-INF/train.csv"));
```

Comment:
I would make this a private static final constant

Reply:
It won't let me make the scanner static since getServletContext() is not a static function. Is there a workaround for this or should I just keep it as is?

Reply:
ah right then just make it not static

Reply:
So whenever I try to move the Scanner object out of the init() function, I get a bunch of errors including "Servlet Config has not been initialized" which eventually cause a null pointer exception. So I just kept the initialization of the Scanner object in the init() function, but made it final. I don't really understand these errors though, do you know the cause for them?

Reply:
My mistake, this can be left here.  Please disregard!